### DISCUSSION TITLE
Much more ouput in Messages pane after updating MiKTeX

### DISCUSSION THREAD

**Discussion by octaeder** [texstudio-org/texstudio#3484] [Q&A] [answered]
I would like to ask how normal behavior looks like.
I start a compile&view with LuaLatex. For a long time the message pane showed like this (the additional time display was my PR #3161 half a year ago, which I took the picture from):

![grafik](https://github.com/texstudio-org/texstudio/assets/102688820/d10b14dd-0a81-4d66-a12d-ecf3a0e0d231)

But after installing latest MiKTeX I see a long text similar to the log file. The image shows top and bottom:

![grafik](https://github.com/texstudio-org/texstudio/assets/102688820/bc69f07f-65a8-4146-a337-3af0caeea772)

Total number of lines about 3'500. Any comments?

**Comment by muzimuzhi** [comment]
Running LuaLaTeX from basictex on macOS:
![image](https://github.com/texstudio-org/texstudio/assets/6376638/a582fed2-31e9-420b-8771-40fbe555d6be)

**Reply by muzimuzhi** [reply]
BTW I'm using TeXstudio 4.7.2.

**Reply by octaeder** [reply]
I use txs 4.7.3 current dev

**Comment by octaeder** [comment]
so the question is what happened or how can I revert to old behavior?

**Reply by muzimuzhi** [reply]
Does the problem reproduce with engines other than LuaLaTeX and short dummy tex document(s)?

**Reply by octaeder** [reply]
```latex
\documentclass[10pt,a5paper]{article}
\usepackage{mathtools}
\begin{document}
    \begin{alignat*}{2}
        y'(x) &= \frac{dy}{dx} = \frac{d}{dx}y &\;\;\;\text{und}\;\;\; y''(x) &= \frac{d^2 y}{dx^2}
    \end{alignat*}
\end{document}
```
compile with pdflatex. After first (process) line I see in Messages pane:
```
This is pdfTeX, Version 3.141592653-2.6-1.40.25 (MiKTeX 23.12) (preloaded format=pdflatex.fmt)
 \write18 enabled.

entering extended mode
(texstudio_OUPRmi.tex

LaTeX2e <2023-11-01>
L3 programming layer <2024-01-04>
... and so on ...
Output written on texstudio_OUPRmi.pdf (1 page, 38898 bytes).
SyncTeX written on texstudio_OUPRmi.synctex.gz.
Transcript written on texstudio_OUPRmi.log.

Process exited normally
```

**Reply by octaeder** [reply, marked as answer]
The answer is... Build option has been changed. Should be "Never" but is

![grafik](https://github.com/texstudio-org/texstudio/assets/102688820/84399aaf-aa85-4f00-92ec-dee209eab7b6)

Assume changed unintentionally by mouse wheel scrolling. Really random.
Thanks for your support

**Reply by muzimuzhi** [reply]
Ah I never knew this option before.

**Reply by muzimuzhi** [reply]
> Should be "Never" but is ...

BTW it seems the default choice is "Only For User Commands" (value 1), not "Never" (value 0).
https://github.com/texstudio-org/texstudio/blob/ff50e9c3a5a5e511f44c2928ec9e11046076dcdb/src/configdialog.ui#L868-L886
https://github.com/texstudio-org/texstudio/blob/ff50e9c3a5a5e511f44c2928ec9e11046076dcdb/src/configmanager.cpp#L667

**Reply by octaeder** [reply]
thanks, you are right. Interesting: Even with never I see following output:

![grafik](https://github.com/texstudio-org/texstudio/assets/102688820/756e1283-fcfe-4a25-be5b-76c5103ca1ed)

**Reply by muzimuzhi** [reply]
That's the non-conventional stderr of `makeindex`. From `man makeindex`,

```
       -q        Quiet mode; send no messages to stderr.  By default, progress and error
                 messages are sent to stderr as well as to the transcript file.
```

In addition, texstudio joins stderr lines.

I find no flag that prohibits progress messages in stderr, but keeps error messages.

**Reply by octaeder** [reply]
Thanks for clarifying this. First I tried to add `-q` to the build command in txs. But this didn't help. The solution is to add it to the latex command: `\makeindex[options=-q,...]`

**Reply by muzimuzhi** [reply]
> Thanks for clarifying this. First I tried to add `-q` to the build command in txs. But this didn't help.

It didn't help because perhaps you loaded `imakeidx` (inferred from the optional argument of `\makeindex`), which executes `makeindex` from with LaTeX run, hence independent from the `txs:///makeindex` setting.

**Comment by octaeder** [comment]
not sure. I have following in the preamble (now with -q added):
```
\usepackage{indextools}
\makeindex[intoc,options=-q -g -s makeindex.quote.ist]
\makeindex[options=-q,name=...,title=...,intoc]
```

**Reply by muzimuzhi** [reply]
[`indextools`](https://ctan.org/pkg/indextools)

> is a fork of [imakeidx](https://ctan.org/pkg/imakeidx), with new features and fixed bugs.

If `makeindex` is executed by `txs:///makeindex`, then you will see
```
Process started: makeindex "xxx".idx
```
right before "This is makeindex, version ...".

If `makeindex` is executed within a LaTeX run, no "Process started: makeindex ..." is added to Message pane and you may find `runsystem(makeindex ...)` in log.
